agent/agent.c

```diff
@@ -20,23 +20,199 @@
+        (self->edge_interface));
+
+  g_application_release (app);
+
+  /* chain up */
+  G_APPLICATION_CLASS (hwangsae_agent_parent_class)->dbus_unregister (app,
+      connection, object_path);
+}
+
+gboolean
+hwangsae_agent_edge_interface_handle_start (Hwangsae1DBusEdgeInterface * object,
+    GDBusMethodInvocation * invocation, gchar * arg_id, gpointer user_data)
+{
+  ChamgeReturn ret = CHAMGE_RETURN_OK;
+  g_autofree gchar *cmd = NULL;
+  g_autofree gchar *response = NULL;
+  GError *error = NULL;
+  HwangsaeAgent *self = (HwangsaeAgent *) user_data;
+  gchar *uid = NULL;
+
+  cmd =
+      g_strdup_printf ("{\"to\":\"%s\",\"method\":\"streamingStart\"}", arg_id);
+
+  g_debug ("hwangsae_agent_edge_interface_handle_start, cmd %s", cmd);
+
+  g_object_get (self->chamge_hub, "uid", &uid, NULL);
+  g_assert_cmpstr (uid, ==, DEFAULT_HUB_UID);
+
+  ret =
+      chamge_node_user_command (CHAMGE_NODE (self->chamge_hub), cmd,
+      &response, &error);
+
+  if (ret != CHAMGE_RETURN_OK) {
+    g_debug ("failed to send user command >> %s\n", error->message);
+  }
+
+  hwangsae1_dbus_edge_interface_complete_start (object, invocation);
+
+  return TRUE;
+}
+
+gboolean
+hwangsae_agent_edge_interface_handle_stop (Hwangsae1DBusEdgeInterface * object,
+    GDBusMethodInvocation * invocation, gchar * arg_id, gpointer user_data)
+{
+  ChamgeReturn ret = CHAMGE_RETURN_OK;
+  g_autofree gchar *cmd = NULL;
+  g_autofree gchar *response = NULL;
+  GError *error = NULL;
+  HwangsaeAgent *self = (HwangsaeAgent *) user_data;
+  gchar *uid = NULL;
+
+  cmd =
+      g_strdup_printf ("{\"to\":\"%s\",\"method\":\"streamingStop\"}", arg_id);
+
+  g_debug ("hwangsae_agent_edge_interface_handle_stop, cmd %s", cmd);
+
+  g_object_get (self->chamge_hub, "uid", &uid, NULL);
+  g_assert_cmpstr (uid, ==, DEFAULT_HUB_UID);
+
+  ret =
+      chamge_node_user_command (CHAMGE_NODE (self->chamge_hub), cmd,
+      &response, &error);
+
+  if (ret != CHAMGE_RETURN_OK) {
+    g_debug ("failed to send user command >> %s\n", error->message);
+  }
+
+  hwangsae1_dbus_edge_interface_complete_stop (object, invocation);
+
+  return TRUE;
+}
+
+static void
+hwangsae_agent_dispose (GObject * object)
+{
+  HwangsaeAgent *self = HWANGSAE_AGENT (object);
+
+  g_clear_object (&self->manager);
+  g_clear_object (&self->edge_interface);
+  g_clear_object (&self->chamge_hub);
+
+  G_OBJECT_CLASS (hwangsae_agent_parent_class)->dispose (object);
+}
+
 static void
 hwangsae_agent_class_init (HwangsaeAgentClass * klass)
 {
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+  GApplicationClass *app_class = G_APPLICATION_CLASS (klass);
+
+  object_class->dispose = hwangsae_agent_dispose;
+
+  app_class->dbus_register = hwangsae_agent_dbus_register;
+  app_class->dbus_unregister = hwangsae_agent_dbus_unregister;
 }
 
 static void
 hwangsae_agent_init (HwangsaeAgent * self)
 {
+  ChamgeReturn ret;
+  gchar *uid = NULL;
+
+  self->chamge_hub = chamge_hub_new_full (DEFAULT_HUB_UID, DEFAULT_BACKEND);
```

Comment:
The ID should be able to be set by configuration. 
Can you add a comment on the top of this line not to forget later?

Reply:
Sure.